### PR TITLE
First draft of exec-env appending to user-agent

### PR DESCRIPTION
**Pull request by stealthycoin**
Attach the environment variable AWS_EXECUTION_ENV to the end of the user agent.

cc @kyleknap @JordonPhillips @jamesls

### CODE REVIEW

File: botocore/args.py [hunk 1]
```diff
@@ -107,6 +108,8 @@ def compute_client_args(self, service_model, client_config,
                 user_agent = client_config.user_agent
             if client_config.user_agent_extra is not None:
                 user_agent += ' %s' % client_config.user_agent_extra
+        if os.environ.get('AWS_EXECUTION_ENV') is not None:
+            user_agent += ' exec-env/%s' % os.environ.get('AWS_EXECUTION_ENV')
```
Comment (stealthycoin) [inline botocore/args.py]: Should this still be appended if a custom client_config is provided?

File: tests/unit/test_args.py [hunk 2]
```diff
@@ -125,3 +126,31 @@ def test_max_pool_from_client_config_forwarded_to_endpoint_creator(self):
...
+
+    def test_execution_env_append(self):
+        with mock.patch('os.environ.get') as m:
+            m.return_value = 'FooEnv'
+            result = self.args_create.compute_client_args(mock.MagicMock(),
+                                                          None,
+                                                          mock.MagicMock(),
+                                                          'us-west-2',
+                                                          '', False, {})
+            self.assertNotEqual(re.match('.+ exec-env/FooEnv$',
```
Comment (JordonPhillips) [inline tests/unit/test_args.py]: I would prefer not to use regex. You can use `.endswith` to achieve what you want.

File: tests/unit/test_args.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 import botocore.config
 from tests import unittest
 import mock
+import re
```
Comment (stealthycoin) [inline tests/unit/test_args.py]: I took this out... in case there are more changes I didn't push it yet.

File: botocore/session.py [hunk 4]
```diff
@@ -472,6 +472,9 @@ def user_agent(self):
                                           platform.release())
         if self.user_agent_extra:
             base += ' %s' % self.user_agent_extra
+        if os.environ.get('AWS_EXECUTION_ENV') is not None:
```
Comment (kyleknap) [inline botocore/session.py]: I think we would want to move this before the user_agent_extra.

File: botocore/session.py [hunk 5]
```diff
@@ -472,6 +472,9 @@ def user_agent(self):
...
+            base += ' exec-env/%s' % os.environ.get('AWS_EXECUTION_ENV')
```
Comment (kyleknap) [inline botocore/session.py]: Also it would be good to add this to the doc string about the ``user_agent`` as all of the other components are documented..

**Pull request merged**